### ISSUE TITLE
`avoid_catching_errors` should also trigger on catch-all.

### ISSUE THREAD

**Issue by HosseinYousefi** [dart-lang/linter#5018]
Converting

```dart
try {
  foo();
} on Error {
  
}
```

to

```dart
try {
  foo();
} catch (_) {
  
}
```

"fixes" the `avoid_catching_errors`. But the replacement is probably even worse!

**Comment by srawlins** [comment]
In the second example, one is not _explicitly_ catching errors, so I think it's ok to not report there. The catch clause in the second example is reported by https://dart.dev/tools/linter-rules/avoid_catches_without_on_clauses.

**Comment by HosseinYousefi** [comment]
Fair enough.
